core/src/main/java/ar/com/wolox/wolmo/core/util/ImageProvider.kt

```diff
@@ -0,0 +1,247 @@
+package ar.com.wolox.wolmo.core.util
+
+import android.content.Context
+import android.content.Intent
+import android.graphics.Bitmap
+import android.graphics.BitmapFactory
+import android.net.Uri
+import android.os.Environment
+import android.provider.MediaStore
+import androidx.annotation.IntRange
+import androidx.annotation.StringDef
+import androidx.annotation.StringRes
+import androidx.fragment.app.Fragment
+import ar.com.wolox.wolmo.core.di.scopes.ApplicationScope
+import java.io.ByteArrayOutputStream
+import java.io.File
+import java.io.IOException
+import javax.inject.Inject
+
+/**
+ * Utils class to manipulate images, through {@link Bitmap}s or their corresponding {@link Uri}, and
+ * for retrieving pictures from gallery/taking them from the camera.
+ */
+@ApplicationScope
+class ImageProvider @Inject constructor(val context: Context, val toastFactory: ToastFactory, val wolmoFileProvider: WolmoFileProvider) {
```

Comment:
If this is gonna be `ApplicationScope`d we should move the methods to it, instead of the `companion object`.